src/studio/src/designer/frontend/dashboard/common/components/MakeCopyModal.test.tsx

```diff
@@ -1,76 +1,93 @@
+      screen.queryByText(/dashboard\.field_cannot_be_empty/i),
+    ).not.toBeInTheDocument();
+
+    expect(postSpy).toHaveBeenCalledWith(
+      `${window.location.origin}/designer/api/v1/repos/copyapp?org=${org}&sourceRepository=${app}&targetRepository=${repoName}`,
+    );
   });
 
-  it('should show error message when clicking confirm and name is too long', () => {
-    const component = mountComponent();
-    expect(component.find(ErrorPopover).prop('anchorEl')).toBe(null);
-
-    const cloneBtn = component.find('button#clone-button');
-    const input = component.find('#new-clone-name-input').hostNodes();
+  it('should show error message when clicking confirm and name is too long', async () => {
+    render();
 
-    input.simulate('change', {
-      target: { value: 'this-new-name-is-way-too-long-to-be-valid' },
+    expect(
+      screen.queryByText(/dashboard\.service_name_is_too_long/i),
+    ).not.toBeInTheDocument();
+    const confirmButton = screen.getByRole('button', {
+      name: /dashboard\.make_copy/i,
     });
-    cloneBtn.simulate('click');
-
-    expect(component.find(ErrorPopover).prop('anchorEl')).not.toBe(null);
-    expect(component.find('button#clone-button').exists()).toBe(true);
-    expect(component.find('button#cancel-button').exists()).toBe(true);
+    const inputField = screen.getByRole('textbox');
+    await user.type(inputField, 'this-new-name-is-way-too-long-to-be-valid');
+    await user.click(confirmButton);
+    expect(
+      screen.getByText(/dashboard\.service_name_is_too_long/i),
+    ).toBeInTheDocument();
   });
 
-  it('should show error message when clicking confirm and name contains invalid characters', () => {
-    const component = mountComponent();
-    expect(component.find(ErrorPopover).prop('anchorEl')).toBe(null);
-
-    const cloneBtn = component.find('button#clone-button');
-    const input = component.find('#new-clone-name-input').hostNodes();
+  it('should show error message when clicking confirm and name contains invalid characters', async () => {
+    render();
 
-    input.simulate('change', {
-      target: { value: 'this name is invalid' },
+    expect(
+      screen.queryByText(/dashboard\.service_name_has_illegal_characters/i),
+    ).not.toBeInTheDocument();
+    const confirmButton = screen.getByRole('button', {
+      name: /dashboard\.make_copy/i,
     });
-    cloneBtn.simulate('click');
-
-    expect(component.find(ErrorPopover).prop('anchorEl')).not.toBe(null);
-    expect(component.find('button#clone-button').exists()).toBe(true);
-    expect(component.find('button#cancel-button').exists()).toBe(true);
+    const inputField = screen.getByRole('textbox');
+    await user.type(inputField, 'this name is invalid');
+    await user.click(confirmButton);
+    expect(
+      screen.getByText(/dashboard\.service_name_has_illegal_characters/i),
+    ).toBeInTheDocument();
   });
 });
 
-const mountComponent = () => {
+const render = (props: Partial<IMakeCopyModalProps> = {}) => {
```

Comment:
Denne funksjonen ville jeg bare kalt `renderMakeCopyModal` og plassert den i toppen av fila, eller i det minste før den blir brukt. Da er det litt enklere for nestemann å skjønne direkte hva det er for noe.

Reply:
I'm a bit on the other side of this - I would prefer to see the actual test scenarios first, as they are what I'm interested in when looking at the test. How the component is rendered is less important. Naming it just `render` also makes it easier to copy this and use as a starting point in other areas.

If you feel strongly about it however, I can change it 🙂